azurelinuxagent/common/osutil/redhat.py

```diff
@@ -122,11 +122,12 @@ def get_nm_controlled(self):
         filepath = "/etc/sysconfig/network-scripts/ifcfg-{0}".format(ifname)
         nm_controlled_cmd = ['grep', 'NM_CONTROLLED=', filepath]
         try:
-            result = shellutil.run_command(nm_controlled_cmd, log_error=False).rstrip()
+            result = shellutil.run_command(nm_controlled_cmd, log_error=False, encode_output=False).rstrip()
```

Comment:
why is encide_output needed?

Reply:
If encode_output is True the stdout is returned as a string, otherwise it is returned as a bytes object.

When True, the result is unicode (u'no'). It was easier to do the comparison if encode_output=False and the result is a string